packages/flutter_tools/templates/app_shared/linux.tmpl/CMakeLists.txt.tmpl

```diff
@@ -105,9 +105,9 @@ if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 endif()
 
 # Start with a clean build bundle directory every time.
-install(CODE "
```

Comment:
@stuartmorgan This is not yet a request to review this PR (lets merge the parent PR first).

But I do have a question, in https://github.com/flutter/flutter/pull/58174/files, we added these lines.
Is it intentional that the bundle dir is only deleted at the install-phase? 

In order to be able to to let `tool_backend.dart` with its `flutter assemble` call do the bundling, this folder needs to be deleted _before_ executing tool_backend.dart, so for example immediately when the CMake script is executed. Any concerns or alternative ideas?

Reply:
> Is it intentional that the bundle dir is only deleted at the install-phase?

Yes, because that's the phase that copies to it, and deleting it when we aren't going to make a new copy would not be good.

> In order to be able to to let `tool_backend.dart` with its `flutter assemble` call do the bundling, this folder needs to be deleted _before_ executing tool_backend.dart, so for example immediately when the CMake script is executed. Any concerns or alternative ideas?

Can that step copy to the intermediate directory instead of the final bundle directory? That's supposed to just be an assemblage of things that are elsewhere in the build output, to make it self-contained.

Reply:
> Can that step copy to the intermediate directory instead of the final bundle directory? That's supposed to just be an assemblage of things that are elsewhere in the build output, to make it self-contained

I was not able to find a copy command in cmake that copies all the files inside a directory to inside another directory. One can make a list with GLOB, and copy all the files in the list. But that doesn't work because the folder is empty during the "initial-phase":

```
file(GLOB NATIVE_ASSET_FILES "${NATIVE_ASSETS_DIR}/*.so") // evaluated during initial phase
foreach(native_asset_file ${NATIVE_ASSET_FILES})
  install(FILES "${native_asset_file}" // evaluated during install phase
    DESTINATION "${INSTALL_BUNDLE_LIB_DIR}"
    COMPONENT Runtime)
```

So I believe the one way to make it work then is to invoke a dart/shell script that does the copying. Unless there's a better solution.

Reply:
Ah, using the install directory command works, _as long as one ends with a slash_. (I was having trouble with just using the install command with directories earlier because it would create an extra directory making it bundle/lib/linux/libfoo.so instead of bundle/lib/libfoo.so).

```
set(NATIVE_ASSETS_DIR "${PROJECT_BUILD_DIR}native_assets/linux/")
install(DIRECTORY "${NATIVE_ASSETS_DIR}"
   DESTINATION "${INSTALL_BUNDLE_LIB_DIR}"
   COMPONENT Runtime)
```